### DISCUSSION TITLE
How to setup miniDexed for editing

### DISCUSSION THREAD

**Discussion by ichingxs4all** [probonopd/MiniDexed#622] [Q&A] [answered]
I am puzzeled by how to setup miniDexed to be able to edit the parameters through a DX7 editor like Dexed or Synthmata.

I am able to play notes, change presets but when I change other parameters ..nothing happens. 
I tried usb midi  host setup, hardware midi and usbgadget mode midi.
I set the DX7 channel to 1 and made sure it sends it to the right midi port. No luck.
I can see the sysex messages beeing send to miniDexed by using a midi sniffer.

What am I missing ?

**Comment by Banana71** [comment, marked as answer]
Hello @ichingxs4all,
There are so many ways to connect the devices. I'll describe how I connect my miniDexed to edit sounds.
I connected the keyboard classic MIDI Out to the MIDI In of my UMC1820 audio interface. The MIDI Out from the UMC1820 goes into the miniDexed MIDI Input.
I configured Dexed as shown in the pictures:
It is important to set the "DX7" MIDI out interface and the MIDI channel under the PARM button (Dexed Parameters, Pic.2) !!

![Dexed Audio Device MIDI](https://github.com/probonopd/MiniDexed/assets/104296205/d67625bd-437a-4c3a-9d89-ea771bf8f921)

![Dexed Parameters](https://github.com/probonopd/MiniDexed/assets/104296205/acbb9547-b91a-4b2b-a56b-e3771cb4ba00)

**Reply by Lonolo** [reply]
_"I tried usb midi host setup, hardware midi and usbgadget mode midi."_

I've only been successful with hardware MIDI, 

Is this possible with USB-MIDI host and/or Gadget mode?

**Reply by diyelectromusic** [reply]
The actual means of getting a MIDI connection should be independent of what you can then do over MIDI, so it shouldn't matter if you are using serial MIDI, USB MIDI as a host or USB MIDI as a device (gadget).

Kevin

**Reply by Lonolo** [reply]
I can connect, send note and CC, but no param changing using USB: Gadget or host. 

Only works with hardware MIDI for me.

Was wondering if anyone else had success.

Your USB Gadget accepts param changes?

(Edited)

...didn't think so.

So apparently it DOES matter!

**Reply by diyelectromusic** [reply]
Apparently it does!  I've replied to your other discussion with a theory, but I'll need to wade through more of circle's MIDI handling to work out how SysEx gets through the stack to MiniDexed via USB MIDI...

Watch this space!

Kevin

**Comment by ichingxs4all** [comment]
Thnx Bananana71 !
It's working now.
I made sure all the MIDI in and OUT are coming from the same MIDI interface port.
I have a complex midi setup with a load of different midi interfaces.
